### ISSUE TITLE
Suggest replacing `.GetValueOrDefault()` with `?? default`

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn#55147]
**Brief description:**

`?? 0`, `?? false`, and `?? default` are more idiomatic forms than `.GetValueOrDefault()`, and the resulting IL is exactly the same (https://github.com/dotnet/roslyn/issues/22800).

It would be nice to have a fix-all which adds and removes parentheses.

**Languages applicable:**

C#, not sure about VB.

**Code example that the analyzer should report:**

This is similar to code I keep seeing in the wild:

```cs
class C
{
    int M(string p) => (p?.Length).GetValueOrDefault();
}
```

💡 Use null-coalescing operator [should remove parens]:

```cs
class C
{
    int M(string p) => p?.Length ?? default;
}
```

Here's a case where parens should be added:

```cs
class C
{
    int M(int? p) => p.GetValueOrDefault() + 1;
}
```

💡 Use null-coalescing operator:

```cs
class C
{
    int M(int? p) => (p ?? default) + 1;
}
```

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by ryanbrandenburg** [comment]
Design Meeting Notes:
- Add refactoring that goes both ways (with some exceptions like use cancellationToken.None instead of `default`).
- Also needs to handle being passed the default value (because `GetValueOrDefault` takes a parameter).
